components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1029,15 +1029,18 @@ where
                 break;
             }
 
-            let expect_index = self.apply_state.get_applied_index() + 1;
-            if expect_index != entry.get_index() {
-                panic!(
-                    "{} expect index {}, but got {}, ctx {}",
-                    self.tag,
-                    expect_index,
-                    entry.get_index(),
-                    apply_ctx.tag,
-                );
+            // the applied index of witness is not continuous, skip index check
+            if !self.peer.is_witness {
```

Comment:
If there are multiple admin commands, and the first one changes it from witness to non-witness, will it panic at L1035?

Reply:
It should be considered in the switch witness PR having a field to indicate the peer is `unavailable`. Add a TODO here.

Reply:
Yes, this situation has been dealt with in another PR